vms/platformvm/blocks/builder/builder.go

```diff
@@ -147,7 +147,13 @@ func (b *builder) AddUnverifiedTx(tx *txs.Tx) error {
 		return err
 	}
 
-	if err := b.Mempool.Add(tx); err != nil {
+	// Retrieved Preferred timestamp to make sure we correctly handle
+	// Continuous Staking fork in the mempool
+	preferred, err := b.Preferred()
+	if err != nil {
+		return err
+	}
+	if err := b.Mempool.Add(tx, preferred.Timestamp()); err != nil {
```

Comment:
Continuous staking fork will change the way txs are store in mempool. Once activated, there won't be need anymore to distinguish stakers txs and drop those with start time too far ahead in the future.
Currently the easier way to let mempool handle the change in tx classification is to pass timestamp of preferred block upon add.